From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 12 Sep 2015 21:04:09 +0200
Subject: Be pessimistic about daemon startup

Only return with true if we managed to get to the main loop. Otherwise
return with an error. This fixes lots of situations where the daemon
claimed to start but didn't (e.g. when not being able to connect
to the database or to write to the attachment folder).
---
 provider/server/ECServer.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/provider/server/ECServer.cpp b/provider/server/ECServer.cpp
index 8ab4920..40e3044 100644
--- a/provider/server/ECServer.cpp
+++ b/provider/server/ECServer.cpp
@@ -1164,7 +1164,9 @@ static int running_server(char *szName, const char *szConfig,
 	if (!daemonize)
 		setsid();
 	unix_create_pidfile(szName, g_lpConfig, g_lpLogger);
-
+	if (unix_runas(g_lpConfig, g_lpLogger)) {
+		goto exit;
+	}
 	mainthread = pthread_self();
 
 	// SIGSEGV backtrace support
